n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,239 @@
+"""
+The evolven3fit_API module provides the necessary tools to evolve a fit 
+using eko and to prepare it for postfit.  
+"""
+
+import pathlib
+import logging
+import numpy as np
+import yaml
+import os
+import glob
+
+from ekobox import gen_theory, gen_op, genpdf, gen_info
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import run_dglap
+from eko import output
+
+from . import utils
+from validphys.loader import Loader
+
+
+logger = logging.getLogger(__name__)
+
+
+def evolve_fit(conf_folder, op_card_dict, t_card_dict, eko_path=None, dump_eko=None):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit
+
+    Parameters
+    ----------
+
+        conf_folder: str or pathlib.Path
+            path to the folder containing the fit
+        op_card_dict: dict
+            user settings for the op_card
+        t_card_dict: dict
+            user settings for the t_card
+        eko_path: str or pathlib.Path
+            path where the eko is stored (if None the eko will be
+            recomputed)
+        dump_eko: str or pathlib.Path
+            path where the eko is dumped (if None the eko won't be
+            stored)
```

Comment:
What is the use case for not having an eko? I thought all theories were supposed to come with one.

Reply:
Yes indeed. However, for the moment the ekos are not there so we want to have something to evolve anyway. Once all the ekos will be ready, we could remove this.

Reply:
Not sure I understand that: Seems to me computing the operator is a time intensive task that you don't want to have popping randomply in here and rather should be done explicitly if needed.

Reply:
@Zaharid all new theories will have their ekos, all old theories wont.

@andreab1997 how long does it take to compute? Because if it is comparable with the cpp version then it is ok, if it takes much longer I would rather have it crash unless asked explicitly to compute it for some reason.

Reply:
For the ekos I tried, if you use at least 4 cores, it takes a reasonable amount of time. For sure more than evolven3fit_cpp, but I would say the same order of magnitude

Reply:
I think this script should not be dealing with old theories. Trying to do that likely implies an user error or at least something that requires special care. So I think we would be better of removing that possibility from here, thereby also making the code simpler.

Reply:
Just to mention that you can use even more than 4 cores without any issue (0 implies to use all the available). However when [this PR](https://github.com/N3PDF/eko/pull/105) will be merged eko should be much faster because the `Q2` grid will not be computed sequentially anymore.